PreMailer.Net/PreMailer.Net/StyleClassApplier.cs

```diff
@@ -15,24 +15,49 @@ public static Dictionary<IElement, StyleClass> ApplyAllStyles(Dictionary<IElemen
 			return elementDictionary;
 		}
 
-		private static IElement ApplyStyles(IElement domElement, StyleClass clazz)
-		{
-			var styles = CssElementStyleResolver.GetAllStyles(domElement, clazz);
+	private static IElement ApplyStyles(IElement domElement, StyleClass clazz)
+	{
+		var originalStyleAttr = domElement.Attributes["style"];
+		var hasImportantInOriginalStyle = originalStyleAttr != null && originalStyleAttr.Value.Contains("!important");
+		
+		var styles = CssElementStyleResolver.GetAllStyles(domElement, clazz);
 
-			foreach (var attributeToCss in styles)
-			{
-				SetAttribute(domElement, attributeToCss);
-			}
+		foreach (var attributeToCss in styles)
+		{
+			SetAttribute(domElement, attributeToCss);
+		}
 
-			var styleAttr = domElement.Attributes["style"];
-			if (styleAttr == null || string.IsNullOrEmpty(styleAttr.Value))
+		if (hasImportantInOriginalStyle)
+		{
+			var currentStyleAttr = domElement.Attributes["style"];
+			if (currentStyleAttr != null && !currentStyleAttr.Value.Contains("!important"))
 			{
-				domElement.RemoveAttribute("style");
+				var parser = new CssParser();
+				var originalStyleClass = parser.ParseStyleClass("inline", originalStyleAttr.Value);
+				
+				var currentStyleClass = parser.ParseStyleClass("inline", currentStyleAttr.Value);
+				
+				foreach (var attr in originalStyleClass.Attributes)
+				{
+					if (attr.Important && !currentStyleClass.Attributes.ContainsKey(attr.Style))
+					{
+						currentStyleClass.Attributes.Merge(attr);
+					}
+				}
+				
+				domElement.SetAttribute("style", currentStyleClass.ToString(emitImportant: true));
 			}
+		}
 
-			return domElement;
+		var styleAttr = domElement.Attributes["style"];
+		if (styleAttr == null || string.IsNullOrEmpty(styleAttr.Value))
+		{
+			domElement.RemoveAttribute("style");
 		}
 
+		return domElement;
+	}
```

Comment:
Why is this needed? Aren't the styles already being merged?

Reply:
Instead of applying this logic here, I think it belongs to `CssElementStyleResolver.GetAllStyles` which is called above.